src/main/kotlin/org/elm/ide/actions/ElmCreateFileAction.kt

```diff
@@ -57,6 +61,17 @@ class ElmCreateFileAction : CreateFileFromTemplateAction(CAPTION, "", ElmFileTyp
     fun testHelperCreateFile(name: String, dir: PsiDirectory): PsiFile? =
             createFile(name, ELM_MODULE_KIND, dir)
 
+    override fun isAvailable(dataContext: DataContext): Boolean {
+        if (!super.isAvailable(dataContext)) return false
+
+        val file = CommonDataKeys.VIRTUAL_FILE.getData(dataContext) ?: return false
+        val project = PlatformDataKeys.PROJECT.getData(dataContext)!!
+        val sourceDirs = project.elmWorkspace.allProjects.asSequence()
```

Comment:
I'm not very familiar with the ElmWorkspace code, so let me know if there's a better way to get his info.

Reply:
Use `findProjectForFile()`

this will do it:
```kotlin
return project.elmWorkspace.findProjectForFile(file) != null
```

Reply:
Yep, that's perfect. The performance is a couple of microseconds.